src/request.ts

```diff
@@ -39,23 +71,54 @@ export default class Request {
   private _fetchWithLogging(url: string, options: RequestInit) : Promise<any> {
     this._logRequest(options.method, url);
     let promise = this._fetch(url, options);
-    promise.then((response : any) => {
+    return promise.then((response : any) => {
       this._logResponse(response['jsonPayload']);
+      return response
     });
-    return promise;
   }
 
   private _fetch(url: string, options: RequestInit) : Promise<any> {
     return new Promise((resolve, reject) => {
+      try {
+        this.modelClass.beforeFetch(url, options)
```

Comment:
One thought: If I want a subclass to ignore before/afterFetch, do we want to support just `null`ing out the method?

``` typescript
class Person extends ApplicationRecord {
  static beforeFetch = null // as opposed to `function(..args: any[]) { }`
}
```

Reply:
It's an interesting an idea, but I think I prefer a generic override unless it becomes an issue. It's more verbose, but less ways to do the same thing.